src/com/yetanalytics/dave/func.cljc

```diff
@@ -568,6 +570,227 @@
+                                                     "https://w3id.org/xapi/dod-isd/verbs/recommended" "recommended"
+                                                     "http://adlnet.gov/expapi/verbs/launched"         "launched"}))
+                          ;; timestamp/stored
+                          (sgen/fmap (fn [i]
+                                       (util/inst->timestamp
+                                        #?(:clj (java.util.Date. i)
+                                           :cljs (js/Date. i))))
+                                     ;; one year span
+                                     (sgen/large-integer*
+                                      {:min 1546300800000
+                                       :max 1577836800000}))
+                          ;; activity id/name tuple
+                          (sgen/fmap (fn [x]
+                                       [(str "https://example.com/activities/" x)
+                                        (str "Activity " x)])
+                                     (sgen/elements ["a" "b" "c" "d" "e" "f"])))))))
+         :args (s/? (s/nilable :learning-path/args)))
+  :ret ::ret/result)
+
+(defrecord LearningPath [state]
+  AFunc
+  (init [this]
+    ;; set/reset to init state
+    (assoc-in this [:state :learners] {}))
+  (relevant? [_ statement]
+    ;; "Should return true if the statement is valid basis data for the func. Otherwise, returns false."
+    ;; -  Not going to support statements with substatements or statement references at this time
+    (let [{:strs [object]} statement
+          object-type (common/get-helper object "objectType")]
+      (case object-type
+        "SubStatement" false
+        "StatementRef" false
+        true)))
+  (accept? [_ statement]
+    ;; "If the func can consider this statement given its current state/data, returns true. Otherwise, returns false."
+    (if (-> statement
+            (common/get-helper "actor")
+            common/parse-actor
+            common/handle-actor
+            not-empty)
+      ;; we have usable identity vs don't have it
+      true
+      false))
+  (step [this statement]
+    ;; "Given a novel statement, update state. Returns the (possibly modified) record."
+    (let [{:keys [timestamp id actor verb object]} (common/parse-statement-simple statement)
+          {:keys [verb-id verb-name]}               verb
+          relevant-actor-info                       (common/handle-actor actor)
+          many-actor-agents?                        (-> relevant-actor-info first vector?)
+          [obj-name id-or-members]                  (common/handle-object object)
+          obj-member-names                          (when (vector? id-or-members)
+                                                      (apply str (interpose "," (map first id-or-members))))
+          object-display                            (or obj-member-names obj-name)]
+      (if many-actor-agents?
+        (do
```

Comment:
@FeLungs this conditional branch doesn't do anything as far as I can tell. It just `doseq`s over something and then returns `this`, unmodified. Remember that clojure `defrecord`s (as opposed to `deftypes`) are immutable and can be thought of inside methods as a clojure map.